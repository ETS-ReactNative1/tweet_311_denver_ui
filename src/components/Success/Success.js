import React from 'react';
import { StyleSheet, Text, View, Dimensions, TouchableOpacity, Image } from 'react-native';

const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

export const Success = ({ navigation }) => {

  return(
    <View style={styles.successContainer}>
      <Text style={styles.headerText}>Success</Text>
      <Text style={styles.messageText}>Form submitted and Tweet Posted</Text>
      <Image style={styles.img} source={require('../../../assets/images/completed-task.png')} />
      <Text style={styles.thanksText}>Thanks!</Text>
      <TouchableOpacity style={styles.homeButton} onPress={ () => navigation.navigate('Home') }>
        <Text style={styles.homeLabel}>Home</Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  successContainer: {
    alignItems: 'center',
    backgroundColor: '#fff',
    flexDirection: 'column',
    flex: 1,
    padding: 15,
    width: '100%',
  },
  headerText: {
    color: '#3976EA',
    fontSize: 50,
    marginTop: 20,
  },
  messageText: {
    color: '#3976EA',
    fontSize: 20,
    marginTop: 20,
  },
  img: {
    marginLeft: 'auto',
    marginRight: 'auto',
    marginRight: 35,
    height: 250,
    marginTop: 50,
    width: 250,
  },
  thanksText: {
    color: '#3976EA',
    fontSize: 20,
    marginTop: 40,
  },
  homeButton: {
    alignItems: 'center',
    backgroundColor: '#3976EA',
    borderRadius: 20,
    height: 40,
    justifyContent: 'center',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 50,
    width: '50%',
  },
  homeLabel: {
    color: '#FFFFFF',
    fontSize: 20
  }
});

export default Success;