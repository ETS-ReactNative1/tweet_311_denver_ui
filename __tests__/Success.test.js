import React from 'react';
import renderer from 'react-test-renderer';

import Success from '../src/components/Success/Success';

test('renders success correctly', () => {
  const tree = renderer.create(<Success />).toJSON();
  expect(tree).toMatchSnapshot();
});
