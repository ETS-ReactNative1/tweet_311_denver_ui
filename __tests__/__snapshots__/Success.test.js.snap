// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Success correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3976EA",
        "fontSize": 50,
        "fontWeight": "bold",
        "marginTop": 20,
      }
    }
  >
    Success!
  </Text>
  <Image
    source={1}
    style={
      Object {
        "height": 250,
        "marginRight": -60,
        "marginTop": 35,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#3976EA",
        "fontSize": 24,
        "marginTop": 30,
      }
    }
  >
    Denver might email for more info
  </Text>
  <Text
    style={
      Object {
        "color": "#3976EA",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginTop": 25,
      }
    }
  >
    Denver 311 Case-ID:
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginTop": 25,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3976EA",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 50,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Home
    </Text>
  </View>
</View>
`;
